### PR TITLE
Fix stackoverflow issue with java function window context.

### PR DESCRIPTION
**Pull request by cckellogg**
### Motivation

Fixes stack overflow issue with java function window context.

### Modifications

Removed recursion call.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: pulsar-functions/windowing/src/main/java/org/apache/pulsar/functions/windowing/WindowContextImpl.java [hunk 1]
```diff
@@ -117,7 +117,10 @@ public ByteBuffer getState(String key) {
 
     @Override
     public String getUserConfigValue(String key) {
-        return this.getUserConfigValue(key);
+        if (this.context.getUserConfigValue(key).isPresent()) {
+            return this.context.getUserConfigValue(key).get().toString();
```
Comment (jerrypeng) [inline pulsar-functions/windowing/src/main/java/org/apache/pulsar/functions/windowing/WindowContextImpl.java]: This API is actually incorrect. @cckellogg can you change the signature from returning a String to returning an Optional<Object> as this is the case for functions:

https://github.com/apache/pulsar/blob/master/pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/ContextImpl.java#L238

Comment (cckellogg) [reply pulsar-functions/windowing/src/main/java/org/apache/pulsar/functions/windowing/WindowContextImpl.java]: @jerrypeng this uses a different interface https://github.com/apache/pulsar/blob/master/pulsar-functions/api-java/src/main/java/org/apache/pulsar/functions/api/WindowContext.java
maybe the can be consolidated in another pr?

**Comment by jerrypeng** [comment]
rerun java8 tests

**Comment by jerrypeng** [comment]
rerun java8 tests

**Pull request merged**